jury/metrics/ter/ter_for_language_generation.py

```diff
@@ -0,0 +1,227 @@
+        else:
+            output = sb_ter.corpus_score(predictions, references)
+        return {"score": float(output.score / 100), "num_edits": output.num_edits, "ref_length": output.ref_length}
+
+    def _compute_single_pred_single_ref(
+        self,
+        predictions: LanguageGenerationInstance,
+        references: LanguageGenerationInstance,
+        reduce_fn: Callable = None,
+        normalized: bool = False,
+        no_punct: bool = False,
+        asian_support: bool = False,
+        case_sensitive: bool = False,
+    ):
+        return self._compute_ter_score(
+            predictions=predictions,
+            references=references,
+            normalized=normalized,
+            no_punct=no_punct,
+            asian_support=asian_support,
+            case_sensitive=case_sensitive,
+        )
+
+    def _compute_single_pred_multi_ref(
+        self,
+        predictions: LanguageGenerationInstance,
+        references: LanguageGenerationInstance,
+        reduce_fn: Callable = None,
+        normalized: bool = False,
+        no_punct: bool = False,
+        asian_support: bool = False,
+        case_sensitive: bool = False,
+    ):
+        # SacreBleu inherently supports multiple references.
+        return self._compute_ter_score(
+            predictions=predictions,
+            references=references,
+            normalized=normalized,
+            no_punct=no_punct,
+            asian_support=asian_support,
+            case_sensitive=case_sensitive,
+        )
+
+    def _compute_multi_pred_multi_ref(
+        self,
+        predictions: LanguageGenerationInstance,
+        references: LanguageGenerationInstance,
+        reduce_fn: Callable = None,
+        normalized: bool = False,
+        no_punct: bool = False,
+        asian_support: bool = False,
+        case_sensitive: bool = False,
+    ):
+        scores = []
+        avg_num_edits = 0
+        avg_ref_length = 0
+        for preds, refs in zip(predictions, references):
+            pred_scores = []
+            num_edits = []
+            ref_lengths = []
+            for pred in preds:
+                score = self._compute_ter_score(
+                    predictions=pred,
+                    references=refs,
+                    sentence_level=True,
+                    normalized=normalized,
+                    no_punct=no_punct,
+                    asian_support=asian_support,
+                    case_sensitive=case_sensitive,
+                )
+                pred_scores.append(score["score"])
+                num_edits.append(score["num_edits"])
+                ref_lengths.append(score["ref_length"])
+            pred_score = reduce_fn(pred_scores).item()
+            avg_num_edits += sum(num_edits) / len(num_edits)
+            avg_ref_length += sum(ref_lengths) / len(ref_lengths)
+            scores.append(pred_score)
+        return {
+            "score": sum(scores) / len(scores),
```

Comment:
Why not reduce with `reduce_fn` instead of hardcoding a mean?

Reply:
Reduce function is only applied to reduce each multiple pred or multiple ref block, at the end we always average these reduced blocks, otherwise it is not informative with reduce functions like max, min, etc. So you can view it like this with the pseudo-ish code below,

```python
predictions=[[2preds],[3preds]]
references=[[4refs],[2refs]]

blocks = []
for each block in (predictions, references):
    blocks.add(reduce_fn(block))
final_score = average(blocks)
```

There is a gif in [jury v1 blog post](https://medium.com/codable/jury-evaluating-performance-of-nlg-models-730eb9c9999f) you can also see from there (it is with reduce_fn max)